src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js

```diff
@@ -1,515 +1,478 @@
-(function($){
+(function($) {
 
     $.widget("mapbender.mbLegend", {
         options: {
-            autoOpen: true,
-            target: null,
-            noLegend: "No legend available",
-            elementType: "dialog",
-            displayType: "list",
-            checkGraphic: false,
-            hideEmptyLayers: true,
+            autoOpen:                 true,
+            target:                   null,
+            noLegend:                 "No legend available",
+            elementType:              "dialog",
+            displayType:              "list",
+            checkGraphic:             false,
+            hideEmptyLayers:          true,
             generateLegendGraphicUrl: false,
-            showSourceTitle: true,
-            showLayerTitle: true,
-            showGrouppedTitle: true,
-            maxImgWidth: 0,
-            maxImgHeight: 0
+            showSourceTitle:          true,
+            showLayerTitle:           true,
+            showGroupedTitle:         true,
+            maxImgWidth:              0,
+            maxImgHeight:             0
         },
-        model: null,
-        layerTitle: "",
-        sourceTitle: "",
-        grouppedTitle: "",
-        hiddeEmpty: "",
-        _create: function(){
-            if(!Mapbender.checkTarget("mbLegend", this.options.target)) {
+
+        readyCallbacks: [],
+        callback:       null,
+        model:          null,
+
+        /**
+         * Widget constructor
+         *
+         * @private
+         */
+        _create: function() {
+            var widget = this;
+            var options = widget.options;
+            if(!Mapbender.checkTarget("mbLegend", options.target)) {
                 return;
```

Comment:
Don't we have to throw an Error here?

Reply:
@LazerTiberius Do it.

Please configure target for the element.

Bitte konfigurieren Sie das Target für das Element.

Reply:
> Don't we have to throw an Error here?
  Actually, checkTarget will display an error in the right corner and also one in the js console if no target is defined.

Reply:
👍